src/psyclone/domain/lfric/lfric_arg_descriptor.py

```diff
@@ -175,20 +177,49 @@ def __init__(self, arg_type, iterates_over):
         if separator:
             self._validate_vector_size(separator, arg_type)
 
-        # The 2nd arg is an access descriptor. Permitted accesses for each
-        # argument type are dealt with in the related _validate methods.
+        # The 2nd arg for scalars (1st for the old-style scalar metadata)
+        # is the Fortran primitive type of their data (issue #817 will
+        # introduce this for fields and operators, too).
+        # Note: Here we also set internal "offset" argument required to
+        #       support the old and the current argument metadata style.
+        # TODO in #874: Remove support for the old-style metadata and
+        #               raise a ParseError if the 2nd argument is not
+        #               a valid data type.
+        if arg_type.args[1].name in \
```

Comment:
Did you mean that the first entry is the new scalar specification, as the new one needs to have explicit datatype? I modified the condition to check for `self._argument_type == "gh_scalar" and arg_type.args[1].name in LFRicArgDescriptor.VALID_ARG_DATA_TYPES` and raise `ParseError` if not. This will have to be modifed in #817 when fields and operators will have data type as well but I prefer to think about it in that issue.